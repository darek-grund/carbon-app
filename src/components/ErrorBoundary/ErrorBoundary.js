import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Pod from 'carbon-react/lib/components/pod';

export default class ErrorBoundary extends Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static propTypes = {
    children: PropTypes.node.isRequired
  };

  componentDidCatch(error, errorInfo) {
    this.setState({ hasError: true, error, errorInfo });
  }

  render() {
    if (this.state.hasError) {
      return (
        <Pod title='Error Boundary Demo' style={ { marginTop: '40px', marginBottom: '40px' } }>
          <details style={ { whiteSpace: 'pre-wrap' } }>
            {this.state.error && this.state.error.toString()}
            {this.state.errorInfo.componentStack}
          </details>
        </Pod>
      );
    }

    return this.props.children;
  }
}
