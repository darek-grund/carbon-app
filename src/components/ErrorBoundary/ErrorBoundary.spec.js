import React from 'react';
import { mount, shallow } from 'enzyme';
import ErrorBoundary from './ErrorBoundary';

const ComponentWithError = () => <div>Error Component</div>;

// Demonstration of a simple test for the HelloWorld component:
describe('<ErrorBoundary />', () => {
  it('renders as expected', () => {
    const children = <div>children</div>;
    const wrapper = shallow(<ErrorBoundary>{children}</ErrorBoundary>);
    expect(wrapper).toMatchSnapshot();
  });

  it('displays error message when wrapped component throws', () => {
    const wrapper = mount(<ErrorBoundary><ComponentWithError /></ErrorBoundary>);
    wrapper.find(ComponentWithError).simulateError(new Error());
    expect(wrapper.state('hasError')).toBe(true);
  });
});
