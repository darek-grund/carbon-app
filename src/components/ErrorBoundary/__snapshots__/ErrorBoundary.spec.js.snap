// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> renders as expected 1`] = `
<div>
  children
</div>
`;
